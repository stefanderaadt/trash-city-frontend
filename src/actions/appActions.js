export const setScreen = screen => {
    return {
        type: 'SET_SCREEN',
        payload: screen
    };
};
