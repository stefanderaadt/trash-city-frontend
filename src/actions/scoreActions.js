import axios from 'axios';

import store from '../store';

export const setScoreName = name => {
  return {
    type: 'SET_SCORE_NAME',
    payload: name
  };
};

const savingScore = () => {
  return {
    type: 'SAVING_SCORE'
  };
};

const savedScore = score => {
  return {
    type: 'SAVED_SCORE',
    payload: score
  };
};

const savingScoreError = () => {
  return {
    type: 'SAVING_SCORE'
  };
};

export const saveHighScore = () => {
  return dispatch => {
    dispatch(savingScore());

    const {
      game: { time: score },
      level: { levelId },
      score: { name }
    } = store.getState();

    const query = `mutation ($input: ScoreInput) {
      setScore(input: $input) {
        id
      }
    }`;

    const input = { name, score, levelId };

    axios
      .post(`${process.env.API_URL}/graphql`, {
        query,
        variables: {
          input
        }
      })
      .then(res => dispatch(savedScore({ ...input, id: res.data.data.setScore.id })))
      .catch(err => dispatch(savingScoreError(err)));
  };
};

const fetchingHighScores = () => {
  return {
    type: 'FETCHING_HIGH_SCORES'
  };
};

const fetchedHighScores = res => {
  return {
    type: 'FETCHED_HIGH_SCORES',
    payload: res.data.data.highScores
  };
};

const fetchingHighScoresError = () => {
  return {
    type: 'FETCHING_HIGH_SCORES_ERROR'
  };
};

export const fetchHighScores = () => {
  return dispatch => {
    dispatch(fetchingHighScores());

    const {
      level: { levelId }
    } = store.getState();

    const query = `query ($levelId: Int!){
      highScores(levelId: $levelId, limit: 5) {
        id
        name
        score
      }
    }`;

    axios
      .post(`${process.env.API_URL}/graphql`, {
        query,
        variables: {
          levelId
        }
      })
      .then(res => {
        dispatch(fetchedHighScores(res));
      })
      .catch(err => dispatch(fetchingHighScoresError(err)));
  };
};
