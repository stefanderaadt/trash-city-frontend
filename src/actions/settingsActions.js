export function setVolume(v) {
  return {
    type: 'SET_VOLUME',
    payload: v
  };
}
