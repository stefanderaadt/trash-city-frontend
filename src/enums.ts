enum Screens {
  Loading,
  MainMenu,
  LevelMenu,
  SettingsMenu,
  ControlsMenu,
  Game
}

enum Popups {
  None,
  Pause,
  Score,
  Info
}

enum GameEventIds {
  SidewalkArea = 1,
  CivilianSpawnPoint = 2,
  WaterTile = 3,
  FilterSystem = 4,
  CleanerSpawnPoint = 5
}

enum TileLayers {
  Background,
  TopAssets1,
  TopAssets2
}

enum ObjectDirection {
  Up,
  Down,
  Left,
  Right
}

enum Keys {
  ArrowUp = 38,
  ArrowDown = 40,
  ArrowLeft = 37,
  ArrowRight = 39,
  SpaceBar = 32
}

enum Flags {
  FlippedHorizontally = 0x80000000,
  FlippedVertically = 0x40000000,
  FlippedDiagonally = 0x20000000
}

export { Screens, Popups, GameEventIds, ObjectDirection, Keys, TileLayers, Flags };
