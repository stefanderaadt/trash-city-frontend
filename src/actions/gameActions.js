import { Popups } from '../enums.ts';
import { fetchHighScores } from './scoreActions';

export const setScore = score => ({
  type: 'SET_SCORE',
  payload: score
});

export const setTime = time => ({
  type: 'SET_TIME',
  payload: time
});

const pauseGameAction = () => ({
  type: 'PAUSE_GAME'
});

const showPopupAction = popup => ({
  type: 'SHOW_POPUP',
  payload: popup
});

export function showPopup(popup) {
  return dispatch => {
    if (popup === Popups.Score) dispatch(fetchHighScores());
    dispatch(showPopupAction(popup));
  };
}

export const pauseGame = () => {
  return dispatch => {
    dispatch(pauseGameAction());
    dispatch(showPopup(Popups.Pause));
  };
};

const resumeGameAction = () => ({
  type: 'RESUME_GAME'
});

export const hidePopup = () => ({
  type: 'HIDE_POPUP'
});

export const resumeGame = () => {
  return dispatch => {
    dispatch(resumeGameAction());
    dispatch(hidePopup());
  };
};
