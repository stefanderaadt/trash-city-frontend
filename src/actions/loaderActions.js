import { Screens } from '../enums.ts';
import { setScreen } from './appActions';

export const startLoading = () => {
  return {
    type: 'START_LOADING'
  };
};

export const onLoading = percentage => {
  return {
    type: 'ON_LOADING',
    payload: percentage
  };
};

const onLoaded = () => {
  return {
    type: 'ON_LOADED'
  };
};

export const loadingCompleted = () => {
  return dispatch => {
    dispatch(onLoaded());
    dispatch(setScreen(Screens.MainMenu));
  };
};
