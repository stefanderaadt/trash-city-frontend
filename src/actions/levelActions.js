import { Popups } from '../enums.ts';
import { showPopup } from './gameActions';

function setLevelData(level) {
  return {
    type: 'SET_LEVEL_DATA',
    payload: level
  };
}

export const levelInit = level => {
  return dispatch => {
    dispatch(setLevelData(level));
    dispatch(showPopup(Popups.Info));
  };
};
