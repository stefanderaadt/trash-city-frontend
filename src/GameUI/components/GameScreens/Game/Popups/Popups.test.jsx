/* eslint-disable no-console */
import React from 'react';
import { render, mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import { Popups, Screens } from '../../../../../enums.ts';
import PopupsComponent from './index';
import { setScreen } from '../../../../../actions/appActions';
import { setScoreName, saveHighScore } from '../../../../../actions/scoreActions';

const mockStore = configureStore([]);

jest.mock('../../../../../Game/Game.ts', () => {
  return {};
});

describe('Popups tests', () => {
  test('Popups rendering no popups when `Popups.None` is set', () => {
    const store = mockStore({
      game: {
        currentPopup: Popups.None
      }
    });

    const wrapper = render(
      <Provider store={store}>
        <PopupsComponent />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('Popups rendering pause Popup correctly', () => {
    const store = mockStore({
      game: {
        currentPopup: Popups.Pause
      }
    });

    const wrapper = render(
      <Provider store={store}>
        <PopupsComponent />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('Popups rendering score Popup correctly', () => {
    const store = mockStore({
      game: {
        score: 0,
        time: 300,
        currentPopup: Popups.Score
      },
      score: {
        scoreName: 'nameeee',
        highScores: [
          { id: '1245215215g', name: 'Stefan', score: 182 },
          { id: '12455215g', name: 'Naam', score: 12 }
        ],
        fetchingHighScores: false,
        fetchedHighScores: true,
        savingScore: false,
        savedScore: false
      }
    });

    const wrapper = render(
      <Provider store={store}>
        <PopupsComponent />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('Popups rendering info Popup correctly', () => {
    const store = mockStore({
      game: {
        currentPopup: Popups.Info
      },
      level: {
        name: 'LEVEL 1',
        goal: 'Level doel'
      }
    });

    const wrapper = render(
      <Provider store={store}>
        <PopupsComponent />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('Popups setPage calling the right action', () => {
    const store = mockStore({
      game: {
        score: 0,
        time: 300,
        currentPopup: Popups.Score
      },
      score: {
        scoreName: 'nameeee',
        highScores: [
          { id: '1245215215g', name: 'Stefan', score: 182 },
          { id: '12455215g', name: 'Naam', score: 12 }
        ],
        fetchingHighScores: false,
        fetchedHighScores: false,
        savingScore: false,
        savedScore: false
      }
    });

    store.dispatch = jest.fn();

    const wrapper = mount(
      <Provider store={store}>
        <PopupsComponent />
      </Provider>
    );
    const button = wrapper.find('.popup-button-mainmenu').first();
    button.simulate('click');

    expect(store.dispatch).toHaveBeenCalledTimes(1);
    expect(store.dispatch).toHaveBeenCalledWith(setScreen(Screens.MainMenu));
  });
});
