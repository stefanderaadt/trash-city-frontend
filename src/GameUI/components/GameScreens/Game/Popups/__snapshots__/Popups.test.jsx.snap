// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popups tests Popups rendering info Popup correctly 1`] = `
<div
  class="game-popup"
>
  <div
    class="game-popup-container"
  >
    <div
      class="info-popup"
    >
      <h1>
        LEVEL 1
      </h1>
      <p>
        Level doel
      </p>
      <div
        class="popup-button-container"
      >
        <div
          class="button popup-button-start"
          role="button"
          tabindex="0"
        >
          Starten
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Popups tests Popups rendering no popups when \`Popups.None\` is set 1`] = `null`;

exports[`Popups tests Popups rendering pause Popup correctly 1`] = `
<div
  class="game-popup"
>
  <div
    class="game-popup-container"
  >
    <div
      class="pause-popup"
    >
      <h1>
        Pauze
      </h1>
      <div
        class="pause-popup-buttons"
      >
        <div
          class="popup-button-container"
        >
          <div
            class="button popup-button-resume"
            role="button"
            tabindex="0"
          >
            Verder spelen
          </div>
        </div>
        <div
          class="popup-button-container"
        >
          <div
            class="button popup-button-mainmenu"
            role="button"
            tabindex="0"
          >
            Hoofdmenu
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Popups tests Popups rendering score Popup correctly 1`] = `
<div
  class="game-popup score-popup"
>
  <div
    class="game-popup-container"
  >
    <div
      class="score-popup-content"
    >
      <h1>
        Gewonnen!
      </h1>
      <hr />
      <div
        class="score-popup-info"
      >
        <p>
          Tijd: 05:00
        </p>
      </div>
      <hr />
      <div
        class="score-popup-scores"
      >
        <div
          class="score-popup-highscores"
        >
          <h2>
            Beste tijden
          </h2>
          <p>
            1: Stefan - 03:02
          </p>
          <p>
            2: Naam - 00:12
          </p>
        </div>
        <div
          class="score-popup-save"
        >
          <p>
            Je hebt een top 5 tijd!
          </p>
          <input
            class="popup-score-name-input"
            placeholder="Naam"
            type="text"
            value="nameeee"
          />
          <div
            class="button popup-button-save-highscore"
            role="button"
            tabindex="0"
          >
            Score opslaan
          </div>
        </div>
      </div>
      <hr />
      <div
        class="popup-button-container"
      >
        <div
          class="button popup-button-mainmenu"
          role="button"
          tabindex="0"
        >
          Hoofdmenu
        </div>
      </div>
    </div>
  </div>
</div>
`;
